Cache/IO.lean

```diff
@@ -77,15 +84,30 @@ def runCmd (cmd : String) (args : Array String) (throwFailure := true) : IO Stri
   if out.exitCode != 0 && throwFailure then throw $ IO.userError out.stderr
   else return out.stdout
 
+def runCurl (args : Array String) (throwFailure := true) : IO String := do
+  runCmd (← getCurl) args throwFailure
+
 def validateCurl : IO Bool := do
+  if (← CURLBIN.pathExists) then return true
   match (← runCmd "curl" #["--version"]).splitOn " " with
   | "curl" :: v :: _ => match v.splitOn "." with
     | maj :: min :: _ =>
-      let (maj, min) := (maj.toNat!, min.toNat!)
-      if maj > 7 then return true
-      if maj == 7 && min >= 70 then
-        if min < 81 then
-          IO.println s!"Warning: recommended `curl` version ≥7.81. Found {v}"
+      let version := (maj.toNat!, min.toNat!)
+      let _ := @lexOrd
+      let _ := @leOfOrd
+      if version >= (7, 81) then return true
+      -- TODO: support more platforms if the need arises
+      if (← runCmd "uname" #["-ms"] false).trim == "Linux x86_64" then
+        IO.println s!"curl is too old; downloading more recent version"
+        IO.FS.createDirAll IO.CACHEDIR
+        -- NOTE: we host only one version of curl, which we assume to be at least as recent
+        -- as the version encoded in `CURLBIN`
+        let _ ← runCmd "curl" #[
+          "https://pp.ipd.kit.edu/~ullrich/tmp/curl-x86_64-linux-static", "-o", CURLBIN.toString]
```

Comment:
Obviously this should be hosted somewhere else, could someone move it to the Azure cache?

Reply:
Should we go with the time-honored tradition of abusing github releases for this?  I can make a leanprover-community/static-curl repo and give you access if you want.

Reply:
Sounds fine too, whatever is simpler I would say

Reply:
Invite sent!

Reply:
Done, and added aarch64 for good measure